src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/AlleleSubsettingUtils.java

```diff
@@ -0,0 +1,421 @@
+     * @return                      Vector of new PL's, ordered accorrding to GenotpyeIterator's ordering
+     */
+    private static double[] subsettedGenotypeLikelihoods(final double[] oldLikelihoods, final int ploidy,
+                                                         final List<Allele> originalAlleles, final List<Allele> newAlleles) {
+        //Since this is a private method, we will *assume* that newAlleles is a subset of originalAlleles
+        // Then we have shortcuts based on two trivial cases:
+        if (newAlleles.size() == originalAlleles.size()) {
+            return oldLikelihoods;  //no subsetting
+        } else if (newAlleles.size() == 1) {
+            return new double[] { oldLikelihoods[0]};   //hom ref is the only genotype
+        }
+
+        final int newPLSize = GLVectorSizeCache.getNumLikelihoodElements(newAlleles.size(), ploidy);
+        final double[] newPLs = new double[newPLSize];
+
+        // fill boolean array stating whether each original allele is present in new mapping
+        final List<Boolean> allelePresent = originalAlleles.stream()
+                .map(a -> newAlleles.contains(a)).collect(Collectors.toList());
+
+        // compute old indices of new alleles in case new allele set is not ordered in the same way as old set
+        // Example. Original alleles: {T*,C,G,A}. New alleles: {G,C}. oldIndexTable = [2,1]
+        final int[] oldIndexTable = newAlleles.stream().mapToInt(a -> originalAlleles.indexOf(a)).toArray();
+        for(final GenotpyeIterator iterator = new GenotpyeIterator(originalAlleles.size(),ploidy); iterator.hasNext(); iterator.next()) {
+            final int[] oldAlleleCounts = iterator.getAlleleCounts();
+            final boolean containsOnlyNewAlleles = IntStream.range(0, originalAlleles.size())
+                    .filter(k -> oldAlleleCounts[k] > 0)
+                    .allMatch(allelePresent::get);
+
+            if (containsOnlyNewAlleles) {
+                final int[] newAlleleCounts = IntStream.range(0, newAlleles.size())
+                        .map(a -> oldAlleleCounts[oldIndexTable[a]]).toArray();
+                final int newIndex = GenotpyeIterator.getLinearIndex(newAlleleCounts, newAlleles.size(), ploidy);
+                newPLs[newIndex] = oldLikelihoods[iterator.getLinearIndex()];
+            }
+        }
+
+        return  newPLs;
+    }
+
+    /**
+     * Create the new GenotypesContext with the subsetted PLs and ADs
+     *
+     * @param originalGs               the original GenotypesContext
+     * @param vc                       the original VariantContext
+     * @param allelesToUse             the actual alleles to use with the new Genotypes
+     * @param assignGenotypes          assignment strategy for the (subsetted) PLs
+     * @return a new non-null GenotypesContext
+     */
+    private static GenotypesContext createDiploidGenotypesWithSubsettedLikelihoods(final GenotypesContext originalGs,
```

Comment:
This class brings in all the non-diploid code I could find.  My impression is that the general ploidy code has been in its own corner of the GATK and doesn't necessarily have identical functionality.  I'll try to unify things more.